python/lsstimport.py

```diff
@@ -37,59 +40,62 @@
 # resolved by setting dynamical library loading flags.
 RTLD_GLOBAL = -1
 RTLD_NOW = -1
-try:
-    import dl
-    if hasattr(dl, 'RTLD_GLOBAL'):  RTLD_GLOBAL = dl.RTLD_GLOBAL
-    if hasattr(dl, 'RTLD_NOW'):     RTLD_NOW    = dl.RTLD_NOW
-except ImportError:
-    # 64bit linux does not have a dl module...
-    pass
-except SystemError:
-    # ...if it does it should throw a SystemError
-    pass
 
-try:
-    if RTLD_GLOBAL < 0:
-        import lsst64defs
-        RTLD_GLOBAL = lsst64defs.RTLD_GLOBAL   # usually 0x00100
-    if RTLD_NOW < 0:
-        import lsst64defs
-        RTLD_NOW = lsst64defs.RTLD_NOW         # usually 0x00002
-    DLFLAGS = RTLD_GLOBAL|RTLD_NOW
-except ImportError:
-    sys.stderr.write(
-        "Could not import lsst64defs; please ensure the base package has been built (not just setup).\n"
-    )
+# For portability we try a number of different options
+# Unfortunately on a Mac running 2.7 RTLD_NOW is not available anywhere
+options = ('os', 'DLFCN', 'ctypes', 'dl', 'lsst64defs')
+for mod in options:
```

Comment:
I left `dl` in there because that's what the code did previously. I agree that it's pointless given it's deprecated and we won't be using a python old enough and in 32-bit mode for it to ever work.

I'll remove `dl`. I left `ctypes` in because it does give us `RTLD_GLOBAL`. Whether that's useful depends on whether `lsst64defs` should ever be used. I'll address that in another comment.